pinax/messages/views.py

```diff
@@ -72,41 +69,41 @@ def post(self, request, *args, **kwargs):
             return self.form_invalid(form)
 
 
-@login_required
-def message_create(
-    request, user_id=None,
-    template_name="pinax/messages/message_create.html",
-    form_class=None, multiple=False
-):
-    if form_class is None:
-        if multiple:
-            form_class = NewMessageFormMultiple
-        else:
-            form_class = NewMessageForm
-
-    if user_id is not None:
-        user_id = [int(user_id)]
-    elif "to_user" in request.GET and request.GET["to_user"].isdigit():
-        user_id = map(int, request.GET.getlist("to_user"))
-    if not multiple and user_id:
-        user_id = user_id[0]
-    initial = {"to_user": user_id}
-    if request.method == "POST":
-        form = form_class(request.POST, user=request.user, initial=initial)
-        if form.is_valid():
-            msg = form.save()
-            return HttpResponseRedirect(msg.get_absolute_url())
-    else:
-        form = form_class(user=request.user, initial=initial)
-    return render_to_response(template_name, {
-        "form": form
-    }, context_instance=RequestContext(request))
-
-
-@login_required
-@require_POST
-def thread_delete(request, thread_id):
-    qs = Thread.objects.filter(userthread__user=request.user)
-    thread = get_object_or_404(qs, pk=thread_id)
-    thread.userthread_set.filter(user=request.user).update(deleted=True)
-    return HttpResponseRedirect(reverse("messages_inbox"))
+@method_decorator(login_required, name='dispatch')
+class MessageCreateView(CreateView):
```

Comment:
Let's just use the `LoginRequiredMixin` instead of `method_decorator` and `login_required` decorator.